lib/tasks/earthworks.rake

```diff
@@ -146,6 +146,15 @@ namespace :earthworks do
                     'earthworks:opengeometadata:index']
   end
 
+  desc "Prune old guest users from the database"
+  task :prune_old_guest_user_data, [:months_old] => [:environment] do |t, args|
+    old_bookmarkless_guest_users_ids = User.guests_without_bookmarks
+                                           .where("users.updated_at < :date", { date: months_old.to_i.months.ago })
```

Comment:
I suppose this could go into a scope as well, I waffled a bit.  Happy to do something else here.